draco/core/io.py

```diff
@@ -757,13 +757,12 @@ def _get_params(self, container, dset):
         """
         # Check if dataset should get truncated at all
         if (self.dataset is None) or (dset not in self.dataset):
-            if dset not in container._dataset_spec or not container._dataset_spec[
-                dset
-            ].get("truncate", False):
+            cdspec = container._class_dataset_spec()
+            if dset not in cdspec or not cdspec[dset].get("truncate", False):
```

Comment:
I think I'm just missing something subtle here. Wouldn't we want to use the instance `_dataset_spec` rather than the class `_dataset_spec` when deciding whether or not to truncate, since that could have been changed by the instance?

Reply:
Ahhh. Of course right after writing this I've noticed the difference with `dataset_spec` and `_dataset_spec`